scalalib/src/scalafmt/ScalafmtWorker.scala

```diff
@@ -65,12 +66,15 @@ private[scalafmt] class ScalafmtWorker {
         .create(this.getClass.getClassLoader)
         .withRespectVersion(false)
 
+      val isScalaFmtConfigExists = os.exists(scalafmtConfig.path)
       val configPath =
-        if (os.exists(scalafmtConfig.path))
+        if (isScalaFmtConfigExists)
           scalafmtConfig.path.toNIO
-        else
-          JPaths.get(getClass.getResource("default.scalafmt.conf").toURI)
-
+        else {
+          val temp = Files.createTempFile("temp", ".scalafmt.conf").toUri
+          JPaths.get(temp) 
+        }
```

Comment:
Can we still try to get the default file as resource, and only fall back to the temp file in case that failed?

Reply:
Agree, but the solution still seems not so elegant. When mill is used , unless customized scala format config file is placed, code always goes to the `create tmp file` branch. The default config file is never used.

I tested it via deleting `.scalafmt.conf` and `default.scalafmt.conf` files, and executing test cases already existed. It workes fine.